src/algebraic_geometry/is_open_comap_C.lean

```diff
@@ -0,0 +1,67 @@
+/-
+Copyright (c) 2021 Damiano Testa. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Damiano Testa
+-/
+import algebraic_geometry.prime_spectrum
+import ring_theory.polynomial.basic
+/-!
+The morphism `Spec R[x] --> Spec R` induced by the natural inclusion `R --> R[x]` is an open map.
+-/
+
+open ideal polynomial prime_spectrum set
+
+namespace algebraic_geometry
+
+namespace polynomial
+
+variables {R : Type*} [comm_ring R]
+
+/-- Given a polynomial `f ∈ R[x]`, `image_of_Df` is the subset of `Spec R` where at least one
+of the coefficients of `f` does not vanish.  We will prove that `image_of_Df` is the image of
+`(zero_locus {f})ᶜ` under the morphism `comap C : Spec R[x] → Spec R`. -/
+def image_of_Df (f : polynomial R) : set (prime_spectrum R) :=
+  {p : prime_spectrum R | ∃ i : ℕ , (coeff f i) ∉ p.as_ideal}
+
+lemma is_open_image_of_Df (f : polynomial R) : is_open (image_of_Df f) :=
+begin
+  rw [image_of_Df, set_of_exists (λ i (x : prime_spectrum R), coeff f i ∉ x.val)],
+  exact is_open_Union (λ i, is_open_basic_open),
+end
+
+lemma mem_compl_zero_locus_iff_not_mem {f : R} {I : prime_spectrum R} :
```

Comment:
This lemma is generic, right? So it should move to the main file on `prime_spectrum`.